extensions/amp-live-list/0.1/amp-live-list.js

```diff
@@ -0,0 +1,378 @@
+        LiveListManager.getMinDataPollInterval());
+
+    const maxItems = this.element.getAttribute('data-max-items-per-page');
+    user.assert(Number(maxItems) > 0,
+        `amp-live-list#${this.liveListId_} must have ` +
+        `data-max-items-per-page attribute with numeric value. ` +
+        `Found ${maxItems}`);
+
+    /** @private @const {number} */
+    this.maxItemsPerPage_ = getNumberOrDefault(maxItems,
+        LiveListManager.getMinDataMaxItemsPerPage());
+
+    /** @private @const {!Object<string, string>} */
+    this.knownChildIds_ = Object.create(null);
+
+    this.manager_.register(this.liveListId_, this);
+
+    this.insertFragment_ = this.win.document.createDocumentFragment();
+
+    this.updateSlot_ = user.assert(
+       this.getUpdateSlot_(this.element),
+       'amp-live-list must have an `update` slot.');
+
+    this.itemsSlot_ = user.assert(
+        this.getItemsSlot_(this.element),
+        'amp-live-list must have an `items` slot.');
+
+    this.updateSlot_.classList.add('-amp-hidden');
+    this.eachChildElement_(this.itemsSlot_, item => {
+      item.classList.add(classes.ITEM);
+    });
+
+    this.validateLiveListItems_(this.itemsSlot_, true);
+
+    this.registerAction('update', this.updateAction_.bind(this));
+  }
+
+  /** @override */
+  update(updatedElement) {
+    const container = this.getItemsSlot_(updatedElement);
+    user.assert(container, 'amp-live-list must have an `items` slot');
+    this.validateLiveListItems_(container);
+    const mutateItems = this.getUpdates_(container);
+
+    console.log(mutateItems.insert);
+    // Insert/new items will be contiguous at the top even though they
+    // weren't in the actual request DOM structure.
+    mutateItems.insert.sort(this.sortByDataSortTime_).forEach(child => {
+      child.classList.add(classes.ITEM);
+      child.classList.add(classes.NEW_ITEM);
+      this.insertFragment_.insertBefore(child,
+          this.insertFragment_.firstElementChild);
+    });
+
+    if (mutateItems.insert.length > 0) {
+      this.updateSlot_.classList.remove('-amp-hidden');
+    }
+  }
+
+  /**
+   * Mutates the current elements dom and compensates for scroll
+   * change if necessary.
+   * @private
+   */
+  updateAction_() {
+    if (this.insertFragment_.childElementCount == 0) {
+      return;
+    }
+
+    // TODO(erwinm): do in place update as well as sorting,
+    // correct insertion, tombstoning etc.
+    this.mutateElement(() => {
+      // Remove the new class from the previously inserted items.
+      this.eachChildElement_(this.itemsSlot_, child => {
+        child.classList.remove(classes.NEW_ITEM);
+      });
+      this.itemsSlot_.insertBefore(this.insertFragment_,
+          this.itemsSlot_.firstElementChild);
+
+      // Hide the update button in case we previously displayed it.
+      this.updateSlot_.classList.add('-amp-hidden');
+      this.viewport_./*REVIEW*/scrollIntoView(this.element);
```

Comment:
Yes. It'd just defer the operation until the next vsync. However, it's better to chain to the `mutateElement`'s promise and call `vsync.mutate` from there. A small hiccup - `BaseElement.mutateElement` has a bug in that it doesn't return the result of `Resources.mutateElement` call, even though it declares it correctly. Please fix.

Thus it'd be:
```
  this.muateElement(() => {...})
      .then(() => {
         vsync.mutate(() => scrollInfoView(...))
      })
```